src/transformStateWithClones.js

```diff
@@ -8,6 +8,49 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const k = [];
+  let stateClone = {
+    ...state,
+  };
+
+  for (const action in actions) {
+    const { type } = actions[action];
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = actions[action];
+
+        stateClone = {
+          ...stateClone,
+          ...extraData,
+        };
```

Comment:
Better to use [Object.assign](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) here